stdlib/messaging/nats/src/main/ballerina/nats/streaming/producer.bal

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public type StreamingProducer client object {
+
+    public function __init(Connection conn, string clientId, string clusterId, StreamingConfig? streamingConfig = ()) {
+        self.createStreamingConnection(conn, clusterId, clientId, streamingConfig);
+    }
+
+    function createStreamingConnection(Connection conn, string clusterId, string clientId,
+    StreamingConfig? streamingConfig) = external;
+
+    # Publish data to a given subject.
+    #
+    # + subject - NATS subject to publish data to
+    # + data - data to publish
+    # + return - `string` value representing the NUID (NATS Unique Identifier) of the published message, if the
```

Comment:
```suggestion
    # + return - `string` value representing the NUID (NATS Unique Identifier) of the published message, if the
```

This description is too long. Shall we break it to smaller phrases and add bullet points?